src/classes/AuthenticationSession.ts

```diff
@@ -32,6 +42,59 @@ export class AuthenticationSession {
     this.promptForPassword();
   }
 
+  public obtainNewAuthTokenUsingRefreshToken(): void {
+    if (!AuthenticationSession.sftpFusionAuthClientId) {
+      logger.error(
+        'Cannot obtain new access token without sftp client id.',
+      );
+      return;
+    }
+
+    if (!AuthenticationSession.sftpFusionAuthClientSecret) {
+      logger.error(
+        'Cannot obtain new access token without sftp client secret.',
+      );
+      return;
+    }
```

Comment:
I appreciate these checks, but I think they're occurring too late, and returning early from a method that returns nothing does not clearly communicate that the service is in an invalid state.

Instead, I'd like to confirm that these environment variables are present during server startup, and exit if they're missing. (Are there any other environment variables like that?) I've previously used the [require-env-variable package](https://www.npmjs.com/package/require-env-variable) to do so, and would suggest that here.

Reply:
+1 to checking immediately during server startup.

As part of that, it might be cleaner to have these values passed as part of a configuration to a constructor (and then have type protection for ensuring they exist rather than having to check in runtime).